NeoMathEngine/src/CPU/CpuMathEngineDnnConv.h

```diff
@@ -33,16 +33,29 @@ constexpr int BlobConvolutionCacheSize = 256 * 1024;
 struct CCpuConvolutionDesc : public CCommonConvolutionDesc {
 	TConvAlgo ForwardAlgo;
 	TConvAlgo BackwardAlgo;
-	std::unique_ptr<CConvolutionDesc> SimdConvolutionDesc;
 
-	CCpuConvolutionDesc( const CBlobDesc& source, const CBlobDesc& result, const CBlobDesc& filter,
+	std::unique_ptr<CConvolutionDesc> SimdConvolutionDesc{};
+
+	enum TMMDesc { TSMMD_Forward_Algo0, TSMMD_Forward_Algo1, TSMMD_Learn_Algo1, /*...*/ TSMMD_Count_ };
+	// The array of matrices multiplication optimization descriptors by enum above
+	mutable CCpuSmallMatricesMultiplyDescsArray<TSMMD_Count_> SmallMatricesMulDescs;
+
+	enum TMMDArray { TSMMDA_Backward_Algo1, TSMMDA_Backward_Algo2, TSMMDA_Learn_Algo2, /*...*/ TSMMDA_Count_ };
+	// The C-array of optimization descriptors arrays, which parametrized by matrix height,
+	// to get access to 1 array, use enum above as indeces.
+	mutable CCpuSmallMatricesMultiplyDescsArray</*Height*/> SmallMatricesMulDescsHeightArrays[TSMMDA_Count_];
```

Comment:
Is there scenario when the same desc needs jits for different algos?
It doesn't look like. So... you need 1 desc for forward. And 2 arrays </*Height*/> for backward and for learn.

Reply:
fixed